src/client/components/Organisms/HeroSearch/HeroSearch.jsx

```diff
@@ -0,0 +1,70 @@
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import styles from '../HeroSearch/HeroSearch.module.css';
+//optionally future import { useTheme }? (Dark Theme) and styles below
+
+const HeroSearch = () => {
+  //const { theme } = useTheme();
+  const navigate = useNavigate();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const valueInput = searchParams.get('query');
+
+  const handleSearchFormInput = query => {
+    navigate(`/search?query=${query}`);
+  };
+
+  const handleSubmit = () => {
+    handleSearchFormInput(valueInput);
+  };
+
+  return (
+    <div className={styles.containerImage}>
+      <div className={styles.divFlex}>
+        <div className={styles.soYummy}>
+          <span className={styles.so}> So</span>
+          <span
+            className={styles.yummy}
+            //</span>  <span className={`${styles.yummy} ${theme === 'dark' ? s.darkTheme : ''}`}>
+          >
+            {' '}
+            Yummy
+          </span>
+        </div>
+        <div
+          className={styles.text}
+          //<div className={`${s.text} ${theme === 'dark' ? s.darkTheme : ''}`}>
+        >
+          {' '}
+          "What to cook?" is not only a recipe app, it is, in fact, your
+          cookbook. You can add your own recipes to save them for the future.
+        </div>
+        <form
+          onSubmit={handleSubmit}
+          className={styles.border}
+          //className={`${s.border} ${theme === 'dark' ? s.darkTheme : ''}`}
+        >
+          <input
+            type="text"
+            name="query"
+            onChange={e => {
+              setSearchParams({ query: e.target.value });
```

Comment:
That would be right?:
const handleQueryChange = e => {
setSearchParams({ query:e.target.value});
}

type="text"
name="query"
onChange = {handleQueryChange}

Reply:
I mean that its good idea to pass only variable as a prop because its easier to understand what we are passing instead of using properties of event 💡 
f.ex.
1. declare variable `const variable = e.target.value` then pass it as a prop on `setSearchParams` 👍

Reply:
const handleQueryChange = e => {
  const queryValue = e.target.value; 
  setSearchParams({ query: queryValue }); 
};

<input 
  type="text" 
  name="query" 
  onChange={handleQueryChange}
/>

This is the best I can do :D

Reply:
Perfect ✅